src/AppBundle/Controller/Report/AccountController.php

```diff
@@ -77,6 +78,51 @@ public function editAccountAction(Request $request, $id)
         return $account;
     }
 
+    /**
+     * @Route("/account/{id}/dependent-records")
+     * @Method({"GET"})
+     * @Security("has_role('ROLE_DEPUTY')")
+     */
+    public function accountDependentRecords($id)
+    {
+        $account = $this->findEntityBy(EntityDir\Report\BankAccount::class, $id, 'Account not found'); /* @var $account EntityDir\Report\BankAccount */
+        $this->denyAccessIfReportDoesNotBelongToUser($account->getReport());
+
+        $report = $account->getReport();
+
+        $transferFilter = function($transfer) use ($account) {
+            return $transfer->getFrom() === $account || $transfer->getTo() === $account;
+        };
+
+        $paymentsFilter = function($expense) use ($account) {
+            return $expense->getBankAccount() === $account;
+        };
+
+        $ret = [
+            Report::SECTION_MONEY_TRANSFERS => $report->hasSection(Report::SECTION_MONEY_TRANSFERS)
+                ? count($report->getMoneyTransfers()->filter($transferFilter))
+                : null,
+            'payments' => [
+                Report::SECTION_DEPUTY_EXPENSES => $report->hasSection(Report::SECTION_DEPUTY_EXPENSES)
+                    ?  count($report->getExpenses()->filter($paymentsFilter))
+                    : null,
+                Report::SECTION_GIFTS => $report->hasSection(Report::SECTION_GIFTS)
+                    ? count($report->getGifts()->filter($paymentsFilter))
+                    : null,
+                Report::SECTION_MONEY_IN => $report->hasSection(Report::SECTION_MONEY_IN)
+                   ? count($report->getMoneyTransactionsIn()->filter($paymentsFilter))
+                    : null,
+                Report::SECTION_MONEY_OUT => $report->hasSection(Report::SECTION_MONEY_OUT)
+                    ? count($report->getMoneyTransactionsOut()->filter($paymentsFilter))
+                    : null
+            ]
+        ];
+
+        $ret['paymentsTotal'] = array_sum($ret['payments']);
```

Comment:
I may be wrong but doesnt this return the number of transactions involving the bank account? If so then "paymentsTotal" is misleading as this implies the payment totals have been summed, I think transactionCount might be better?